### ISSUE TITLE
Use protocol handler for snapshots from /saveItems

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#1471]
@fbennett [reported on zotero-dev](https://groups.google.com/d/topic/zotero-dev/c0lvHmfh8aA/discussion) that creating custom documents from translators doesn't work. Looking into it a bit, here's what seems to be the situation:

For some reason we have two different ways of saving snapshots. When saving via `/saveSnaphshot` (for saving a webpage), we pass the innerHTML to the connector server, which stuffs it in a property and then does a save with `importFromDocument()` using a `zotero://connector/[url]` URL, which proxies the stored HTML. Somehow (not quite sure how) that ends up looking to `loadDocuments()` like the real remote URL, and it saves it using the passed HTML and auxiliary files that it downloads. So saved webpages will reflect browser-side changes to the DOM (e.g., made with the inspector or another extension).

By comparison, when an attachment with a document is returned from translators, the connector code [strips it out](https://github.com/zotero/zotero/blob/71bee9cfe74b491f9422129d0e13eb38f92f8f44/chrome/content/zotero/xpcom/translation/translate.js#L170) and only passes `/saveItems` the URL, and Zotero re-downloads the page using `importFromURL()`. This means that 1) snapshots from translators don't reflect browser-side changes, 2) another request is made for the page (which has cookies from the cookie sandbox but still might not reflect the loaded page), and 3) translators aren't able to construct custom documents. In Frank's case, he ends up building a document without a URL, the actual document gets stripped, and the empty URL is set to about:blank, which `importFromURL()` rejects as invalid.

I'm not sure there's a good reason we're not using the protocol-handler mechanism for snapshots from translators, and it seems like that's something we'll want anyway in order to do [simplified snapshots](https://github.com/zotero/zotero-connectors/issues/194) (which we'll probably want to do connector-side, both to take advantage of existing extension code that does this and to have it work when saving to server).

So basically we'd need to adjust `/saveItems` to handle snapshot data the same way as `/saveSnapshot` (which is good, because /saveSnapshot can go away once we have a [generic webpage translator](https://github.com/zotero/translators/issues/1092)) and then have the connector stop stripping documents from translators and send the document HTML instead [as it does](https://github.com/zotero/zotero-connectors/blob/30d574267d36fd8af311a50a72b8057186c05c7c/src/common/inject/inject.jsx#L428) for webpages.

**Comment by dstillman** [comment]
(We'll have to do this in an Electron-compatible way going forward, so the protocol handler may not be the right approach.)
